src/main/scala/com/cloudera/sparkts/DateTimeIndex.scala

```diff
@@ -271,6 +271,160 @@ class IrregularDateTimeIndex(
   }
 }
 
+/**
+ * An implementation of DateTimeIndex that holds a hybrid collection of DateTimeIndex
+ * implementations.
+ *
+ * Indices are assumed to be sorted and disjoint such that
+ * for any two consecutive indices i and j: i.last < j.first
+ *
+ */
+class HybridDateTimeIndex(
+    val indices: Array[DateTimeIndex],
+    val dateTimeZone: ZoneId = ZoneId.systemDefault())
+  extends DateTimeIndex {
+
+  private val sizeOnLeft: Array[Int] = {
+    indices.init.scanLeft(0)(_ + _.size)
+  }
+
+  override def slice(interval: Interval): HybridDateTimeIndex = {
+    slice(ZonedDateTime.ofInstant(interval.getStart, dateTimeZone),
+      ZonedDateTime.ofInstant(interval.getEnd, dateTimeZone))
+  }
+
+  override def slice(start: ZonedDateTime, end: ZonedDateTime): HybridDateTimeIndex = {
```

Comment:
> Does this need to return a HybridDateTimeIndex? Could just return a DateTimeIndex? ~ @sryza 

From the consistency point of view, I think the caller should expect a ```HybridDateTimeIndex``` ... A slice from ```UniformDateTimeIndex``` is a ```UniformDateTimeIndex``` ... A slice from ```IrregularDateTimeIndex``` is an ```IrregularDateTimeIndex``` ... a slice from ```DateTimeIndex``` is a ```DateTimeIndex``` ... Why not a slice from ```HybridDateTimeIndex``` be a  ```HybridDateTimeIndex```?

From another perspective, in parts, a ```HybridDateTimeIndex``` is a collection of other time index types and it could be that a slice is one of these types hence returning the super ```DateTimeIndex``` might be valid. However, I don't think that it'll be always the case that the caller knows in advance the true type of the slice ... it might be hybrid or not. Hence, I favor consistency.

What do you think?